### DISCUSSION TITLE
teslamate backup under unraid

### DISCUSSION THREAD

**Discussion by reddai75** [teslamate-org/teslamate#2912] [Q&A]
Hello people!
I'm struggling to create a manual back-up of all my data, as I want to move teslamate from unraid to a standalone raspberry pi.
I am a total noob, and I couldn't find any solution on the web. I spent the entire day looking for a solution.
I type `docker-compose exec -T database pg_dump -U teslamate teslamate > /afolderofmychoice/teslamate.bck`
and I get `no configuration file provided: not found`

Thanks in advance for your help!

**Comment by natrlhy** [comment]
Are you in the same directory as your docker-compose.yml when issuing that command?

**Comment by reddai75** [comment]
hello thank you for your answer
My first problem is that I can't find docker-compose.yml

**Reply by natrlhy** [reply]
On mine I have it here: /root/docker/teslamate

**Reply by reddai75** [reply]
not on my unraid :(

**Reply by ktsenvolt** [reply]
sudo find / -name 'docker-compose.yml' that will find your file.

**Reply by reddai75** [reply]
thanks a lot, Solved. I could not find docker-compose.yml I probably used the wrong settings with find :)

**Reply by mrmark75** [reply]
Hey reddai75 I am also trying to get 3 years of Teslamate data off my unraid and move it to a standalone machine. I have been unsuccessful care to share how you did it?

**Reply by SBRAndroos** [reply]
I'm beached at this point, anyone got a solution for this? @reddai75  would be glad if you could point me in the right direction.

**Comment by mrmark75** [comment]
> I'm beached at this point, anyone got a solution for this? @reddai75 would be glad if you could point me in the right direction.

I saw your post and I never seen a reply from reddai75 either. I spent countless hours Googling and trying to figure it out on my own how to do this. I was able to get a backup file but I cant tell you if its truly valid because I never finished moving teslamate to another machine and tried restoring it.  Hopeful I can help point you in the right direction. So all the documentation and searches say to use this command: 
 _**docker-compose exec -T database pg_dump -U teslamate teslamate > /afolderofmychoice/teslamate.bck**_  
Unfortunately in Unraid that does not seam to work, I had to take the command and break it down piece by piece. First off I learned the docker-compose is not a valid command in Unraid because it was not setup with a .yml file and there is no good way to add it to Unraid, The command that worked for me is   **_sudo docker exec_** The next part is **_-T database pg_dump_** This was the hardest part to figure out. When using docker exec command you need to use a lower case t not capitol T.  How I figured out what the container name is was to go to your docker tab click on the postgres elephant icon choose console. you will have a # prompt type **_su -_**    then you will have root@container:~# the info after @ is what you need to use, mine is 2e89793c5cde. The last piece of the puzzle is the postgres user and database name. I think if you used the teslamate install documentation or .yml they suggest you use user: teslamate database: teslamate but unfortunately I did not set mine up that way I used user: postgres and database postgres.  You can see what your user name and database is by going to your docker clicking on postgres next to picture and look for POSTGRES_USER: and POSTGRES_DB: the last part is where do you want to save the backup. I created a share on disk1 of my array called Tesla so I am directing the backup there. So now that I have all the pieces this is what my command looks like:  
**_sudo docker exec -t 2e89793c5cde pg_dump -U postgres postgres > /mnt/disk1/Tesla/teslamate.bck_**
 It will take a couple min to run, I have 2 Tesla's and 3 years of data and my file is 3.5gb.  I can then get to the share from another computer and copy the backup file. Like I said earlier I am not 100% sure this is the correct way to back it up but I was able to create a backup so I may be right or close I will not know until I try and restore it. Let me know if you are successful at backing up an restoring. I would like to have someone else to bounce ideas off of.. Good Luck!!  

Some after thoughts that may be important are the fact that I only have one postgres database for teslamate only.  If you have multiple databases in postgres for other things that could change things.  I am also still using Unraid 6.9.2

**Comment by GaryCooper23** [comment]
ssh or console:

```
cd /mnt/user/yourshare
docker compose down
docker compose exec -T database pg_dump -U teslamate teslamate > teslamate_$(date +"%Y_%m_%d_%I_%M_%p").bck
```

Then compose up via gui (Docker Compose Manager)
